src/engine/srv.c

```diff
@@ -773,6 +787,8 @@ dss_start_one_xstream(hwloc_cpuset_t cpus, int tag, int xs_id)
 	} else {
 		dx->dx_main_xs	= (xs_id >= dss_sys_xs_nr) && (xs_offset == 0);
 	}
+	/* See the DSS_XS_IOFW case in sched_ult2xs. */
+	dx->dx_iofw = xs_id >= dss_sys_xs_nr && (!dx->dx_main_xs || dss_tgt_offload_xs_nr == 0);
```

Comment:
minor: Seems you don't regard vos xstream as iofw xstream, so it could be simplified as "dx->dx_iofw = xs_id >= dss_sys_xs_nr && !dx->dx_main_xs".

Reply:
@NiuYawei, I thought about the same when updating the patch last time. Ended up keeping it this way because if we make such a change to `dx_iofw`, then a `dx_iofw` xstream != a DSS_XS_IOFW xstream, potentially making our terminology even more complex (i.e., we currently talk about "I/O forward xstream", "offload xstream", "helper xstream", "DSS_XS_OFFLOAD xstream", etc.---already too confusing).